gossip/src/gossip_service.rs

```diff
@@ -151,20 +151,21 @@ pub fn discover_validators(
     socket_addr_space: SocketAddrSpace,
 ) -> std::io::Result<Vec<ContactInfo>> {
     const DISCOVER_CLUSTER_TIMEOUT: Duration = Duration::from_secs(120);
-    let (_all_peers, validators) = discover(
-        None, // keypair
-        Some(entrypoint),
+    let (_all_peers, validators) = discover_peers(
+        None,
+        &vec![*entrypoint],
         Some(num_nodes),
         DISCOVER_CLUSTER_TIMEOUT,
-        None,             // find_nodes_by_pubkey
-        None,             // find_node_by_gossip_addr
-        None,             // my_gossip_addr
-        my_shred_version, // my_shred_version
+        None,
+        &[],
+        None,
+        my_shred_version,
         socket_addr_space,
     )?;
     Ok(validators)
 }
 
+#[deprecated(since = "3.1.0", note = "use `discover_peers` instead")]
```

Comment:
Unless this is getting backported, we can just remove the old functions.

Reply:
Everything here is behind agave-unstable-api, so we can rip it out if it does not make sense.